uint8 category
float64 xCenter